### PR TITLE
Checkers return models

### PR DESCRIPTION
**Pull request by Alex-Fischman**
This makes the checkers return models so they can be printed in `command.rs`.
It also makes the set checker report convergence when it happens.

### CODE REVIEW

File: bounded/src/bdd.rs [hunk 1]
```diff
@@ -168,21 +167,40 @@ impl Context<'_> {
...
+                        .map(|r| {
+                            let shape = r
+                                .args
+                                .iter()
+                                .map(|s| cardinality(self.universe, s))
+                                .chain([2])
+                                .collect();
+                            Interpretation::new(&shape, |elements| {
+                                valuation
+                                    .value(self.vars[self.indices[r.name.as_str()][elements].0])
```
Comment (tchajed) [inline bounded/src/bdd.rs:196]: I think this expression starting with `self.vars[...]` is a good candidate for a method on `self`.

File: bounded/src/bdd.rs [hunk 2]
```diff
@@ -168,21 +167,40 @@ impl Context<'_> {
...
+                                    as usize
+                            })
+                        })
+                        .collect(),
+                )
+            })
+            .collect()
```
Comment (tchajed) [inline bounded/src/bdd.rs:203]: This function has become quite long. Can you at least factor out this conversion to a separate function (or split further if there's a good way to do so)?

**Comment by tchajed** [comment]
Mostly looks good to me, just saw some complexity that I thought should be factored out.

File: bounded/src/bdd.rs (hunk 1 above)
Comment (Alex-Fischman) [reply bounded/src/bdd.rs:196]: Why? It's not used anywhere else, and `Context::get` does what is normally desired. This is the one place in the code where we convert `BddValuation`s to `Model`s.

Comment (tchajed) [reply bounded/src/bdd.rs]: The reason to create a function in this case would mainly be to name and document it (which is helpful even if used only once). Now that you've factored out `convert_valuation` I agree this isn't as necessary.

**Review by tchajed — APPROVED** [review]

**Pull request merged**